arches/app/models/graph.py

```diff
@@ -496,42 +496,63 @@ def add_function(self, function):
         return function
 
     def add_resource_instance_lifecycle(self, resource_instance_lifecycle):
-        self.resource_instance_lifecycle = models.ResourceInstanceLifecycle(
-            id=resource_instance_lifecycle["id"],
-            name=resource_instance_lifecycle["name"],
+        """
+        Adds a ResourceInstanceLifecycle to this graph
+
+        Arguments:
+        resource_instance_lifecycle -- a dictionary representing a models.ResourceInstanceLifecycle instance
+
+        """
+
+        resource_instance_lifecycle_query = (
+            models.ResourceInstanceLifecycle.objects.filter(
+                pk=resource_instance_lifecycle["id"]
+            )
         )
 
-        resource_instance_lifecycle_states = []
-        for resource_instance_lifecycle_state_json in resource_instance_lifecycle[
-            "resource_instance_lifecycle_states"
-        ]:
-            next_resource_instance_lifecycle_states = (
-                resource_instance_lifecycle_state_json.pop(
-                    "next_resource_instance_lifecycle_states"
-                )
+        self.resource_instance_lifecycle = resource_instance_lifecycle_query.first()
+
+        if not self.resource_instance_lifecycle:
+            self.resource_instance_lifecycle = models.ResourceInstanceLifecycle(
+                id=resource_instance_lifecycle["id"],
+                name=resource_instance_lifecycle["name"],
             )
-            previous_resource_instance_lifecycle_states = (
-                resource_instance_lifecycle_state_json.pop(
-                    "previous_resource_instance_lifecycle_states"
+
+            resource_instance_lifecycle_states = []
+            for resource_instance_lifecycle_state_json in resource_instance_lifecycle[
+                "resource_instance_lifecycle_states"
+            ]:
+                next_resource_instance_lifecycle_states = (
+                    resource_instance_lifecycle_state_json.pop(
+                        "next_resource_instance_lifecycle_states"
+                    )
+                )
+                previous_resource_instance_lifecycle_states = (
+                    resource_instance_lifecycle_state_json.pop(
+                        "previous_resource_instance_lifecycle_states"
+                    )
                 )
-            )
 
-            resource_instance_lifecycle_state = models.ResourceInstanceLifecycleState(
-                **resource_instance_lifecycle_state_json
-            )
+                resource_instance_lifecycle_state = (
+                    models.ResourceInstanceLifecycleState(
+                        **resource_instance_lifecycle_state_json
+                    )
+                )
```

Comment:
I don't think we need to do it in this PR necessarily since it's a pre-existing arches pattern, but one of things I'd like to discuss with other developers soon is our approach to validation. I think it's important that whenever we instantiate a model from scratch we immediately call `full_clean()`, since we're not using model forms or DRF serializers that do it for us.

We have a Farallon brown bag on the books for this later in August 👍